src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -61,8 +62,12 @@ internal static Exception CreateOpenSslCryptographicException()
                 return new CryptographicException();
             }
 
+            // Even though ErrGetError returns ulong (C++ unsigned long), we 
+            // really only expect error codes in the UInt32 range
+            Debug.Assert(error <= uint.MaxValue, "ErrGetError should only return error codes in the UInt32 range.");
```

Comment:
We're unchecked casting error as `int` and making it be ex.HResult.  So we're asserting it fits in 32-bits, so there's no dataloss.